eqt/ui/UIFormWidget.py

```diff
@@ -263,39 +374,78 @@ def getWidgetState(self, widget, role=None):
         elif isinstance(widget, (QtWidgets.QTextEdit, QtWidgets.QPlainTextEdit)):
             widget_state['value'] = widget.toPlainText()
 
+        if role == 'field':
+            widget_state['widget_number'] = self.widget_number_dictionary[name]
         return widget_state
 
-    def applyWidgetState(self, name, state, role=None):
+    def _getNameAndRoleFromNameKey(self, name_key):
```

Comment:
You introduce the special term "key". I found it relatively confusing.

Reply:
I changed "name_key" to be either "name" or "key", as appropriate. The "applyWidgetState" method only works with "name" and "role".